### PR TITLE
case 21485: removing quest-demo specific changes

### PR DESCRIPTION
**Pull request by amerhifi**
https://highfidelity.manuscript.com/f/cases/21485/Merge-quest-demo-into-master

### CODE REVIEW

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 1]
```diff
@@ -380,16 +380,26 @@ void OpenGLDisplayPlugin::customizeContext() {
...
             gpu::ShaderPointer program = gpu::Shader::createProgram(shader::gpu::program::DrawTexture);
-            _simplePipeline = gpu::Pipeline::create(program, scissorState);
-            _hudPipeline = gpu::Pipeline::create(program, blendState);
+#endif 
+           _simplePipeline = gpu::Pipeline::create(program, scissorState);
         }
-
         {
-            gpu::ShaderPointer program = gpu::Shader::createProgram(shader::display_plugins::program::SrgbToLinear);
+#ifdef Q_OS_ANDROID
```
Comment (SamGondelman) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: Should these two also happen if USE_GLES?

File: libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp [hunk 2]
```diff
@@ -168,10 +173,8 @@ bool OculusMobileDisplayPlugin::isHmdMounted() const {
 static void goToDevMobile() {
```
Comment (SamGondelman) [inline libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp]: Is this method still needed?

File: libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp [hunk 3]
```diff
@@ -217,12 +220,12 @@ bool OculusMobileDisplayPlugin::beginFrameRender(uint32_t frameIndex) {
...
-    //      AbstractViewStateInterface::instance()->postLambdaEvent([] {
-    //          goToDevMobile();
-    //      });
-    //  }
+  //  static uint32_t count = 0;
+  //  if ((++count % 1000) == 0) {
+  //      AbstractViewStateInterface::instance()->postLambdaEvent([] {
+  //          goToDevMobile();
+  //      });
+  //  }
```
Comment (SamGondelman) [inline libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp]: can delete

File: scripts/developer/utilities/render/deferredLighting.qml [hunk 4]
```diff
@@ -148,6 +148,27 @@ Rectangle {
             }
         }
         Separator {}          
+        Column {
+            anchors.left: parent.left
+            anchors.right: parent.right 
+            spacing: 5 
+            Repeater {
+                model: [ "MSAA:PrepareFramebuffer:numSamples:4:1"
```
Comment (SamGondelman) [inline scripts/developer/utilities/render/deferredLighting.qml]: Will this work if we're not forward rendering?

File: libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp (hunk 2 above)
Comment (amerhifi) [reply libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp]: its convenient whenever we have to do a demo like yesterday

File: libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp (hunk 3 above)
Comment (amerhifi) [reply libraries/oculusMobilePlugin/src/OculusMobileDisplayPlugin.cpp]: goes along with above function. More of a convinience.

File: interface/src/Application.cpp [hunk 5]
```diff
@@ -9158,6 +9162,15 @@ void Application::enterForeground() {
     auto nodeList = DependencyManager::get<NodeList>();
     nodeList->setSendDomainServerCheckInEnabled(true);
 }
+
+
+void Application::toggleAwayMode(){
```
Comment (jherico) [inline interface/src/Application.cpp]: the problem with this approach is that if some UI element has focus, you're just going to dump the escape into that UI element, and not actually trigger away

File: interface/src/avatar/MyAvatar.cpp [hunk 6]
```diff
@@ -2951,6 +2951,10 @@ void MyAvatar::initAnimGraph() {
         graphUrl = _fstAnimGraphOverrideUrl;
     } else {
         graphUrl = PathUtils::resourcesUrl("avatar/avatar-animation.json");
+
+#if defined(Q_OS_ANDROID) || defined(HIFI_USE_OPTIMIZED_IK)
```
Comment (jherico) [inline interface/src/avatar/MyAvatar.cpp]: why does this need to have a `defined(Q_OS_ANDROID)` component?  If we want this to be only available on Android, we can do that check in the cmake.  This should probably just be `#ifdef HIFI_USE_OPTIMIZED_IK`

File: interface/src/avatar/MySkeletonModel.cpp [hunk 7]
```diff
@@ -33,6 +35,22 @@ Rig::CharacterControllerState convertCharacterControllerState(CharacterControlle
     };
 }
 
+#if defined(Q_OS_ANDROID) || defined(HIFI_USE_OPTIMIZED_IK)
```
Comment (jherico) [inline interface/src/avatar/MySkeletonModel.cpp]: simplify to `#ifdef HIFI_USE_OPTIMIZED_IK`

File: interface/src/avatar/MySkeletonModel.cpp [hunk 8]
```diff
@@ -233,6 +251,12 @@ void MySkeletonModel::updateRig(float deltaTime, glm::mat4 parentTransform) {
             myAvatar->getControllerPoseInAvatarFrame(controller::Action::LEFT_HAND).isValid() &&
             !(params.primaryControllerFlags[Rig::PrimaryControllerType_Spine2] & (uint8_t)Rig::ControllerFlags::Enabled)) {
 
+#if defined(Q_OS_ANDROID) || defined(HIFI_USE_OPTIMIZED_IK)
```
Comment (jherico) [inline interface/src/avatar/MySkeletonModel.cpp]: simplify to `#ifdef HIFI_USE_OPTIMIZED_IK`

File: interface/src/avatar/MySkeletonModel.cpp [hunk 9]
```diff
@@ -243,6 +267,9 @@ void MySkeletonModel::updateRig(float deltaTime, glm::mat4 parentTransform) {
             if (spine2Exists && headExists && hipsExists) {
 
                 AnimPose rigSpaceYaw(myAvatar->getSpine2RotationRigSpace());
+#if defined(Q_OS_ANDROID) || defined(HIFI_USE_OPTIMIZED_IK)
```
Comment (jherico) [inline interface/src/avatar/MySkeletonModel.cpp]: simplify to `#ifdef HIFI_USE_OPTIMIZED_IK`

File: interface/src/avatar/MySkeletonModel.cpp [hunk 10]
```diff
@@ -253,6 +280,9 @@ void MySkeletonModel::updateRig(float deltaTime, glm::mat4 parentTransform) {
                 }
                 generateBasisVectors(up, fwd, u, v, w);
                 AnimPose newSpinePose(glm::mat4(glm::vec4(w, 0.0f), glm::vec4(u, 0.0f), glm::vec4(v, 0.0f), glm::vec4(glm::vec3(0.0f, 0.0f, 0.0f), 1.0f)));
+#if defined(Q_OS_ANDROID) || defined(HIFI_USE_OPTIMIZED_IK)
```
Comment (jherico) [inline interface/src/avatar/MySkeletonModel.cpp]: simplify to `#ifdef HIFI_USE_OPTIMIZED_IK`

File: libraries/animation/src/Rig.cpp [hunk 11]
```diff
@@ -1051,16 +1064,29 @@ void Rig::computeMotionAnimationState(float deltaTime, const glm::vec3& worldPos
...
         t += deltaTime;
 
-        if (_enableInverseKinematics != _lastEnableInverseKinematics) {
-            if (_enableInverseKinematics) {
-                _animVars.set("ikOverlayAlpha", 1.0f);
-            } else {
-                _animVars.set("ikOverlayAlpha", 0.0f);
-            }
+        if (_enableInverseKinematics) {
+            _animVars.set("ikOverlayAlpha", 1.0f);
```
Comment (jherico) [inline libraries/animation/src/Rig.cpp]: all these magic strings should be constants defined somewhere, like `LEFT_HAND_IK_POSITION_VAR`

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp (hunk 1 above)
Comment (jherico) [reply libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: it should probably not be `Q_OS_ANDROID` _or_ `USE_GLES`, but some other option that indicates that we can't support the colorspace output we want on this system...  but that would need to be a runtime check, not a compile time check

File: scripts/developer/utilities/render/deferredLighting.qml (hunk 4 above)
Comment (samcake) [reply scripts/developer/utilities/render/deferredLighting.qml]: this won;t break in deferred

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17720/HighFidelity-Beta-PR15052-faedc61.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17720/HighFidelity-Beta-Interface-PR15052-faedc61.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17720/HighFidelity-Beta-PR15052-faedc61.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17720/HighFidelity-Beta-Interface-PR15052-faedc61.exe

**Comment by amerhifi** [comment]
build this please

**Comment by amerhifi** [comment]
build this please

**Comment by amerhifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17791/HighFidelity-Beta-PR15052-ad69611.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17791/HighFidelity-Beta-Interface-PR15052-ad69611.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17791/HighFidelity-Beta-PR15052-ad69611.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17791/HighFidelity-Beta-Interface-PR15052-ad69611.exe

**Comment by amerhifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17807/HighFidelity-Beta-PR15052-8802eea.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17807/HighFidelity-Quest-Beta-PR15052-8802eea.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17807/HighFidelity-Beta-PR15052-8802eea.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17807/HighFidelity-Beta-Interface-PR15052-8802eea.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17807/HighFidelity-Beta-PR15052-8802eea.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17807/HighFidelity-Beta-Interface-PR15052-8802eea.exe

File: interface/src/Application.cpp (hunk 5 above)
Comment (SamGondelman) [reply interface/src/Application.cpp]: You could maybe fix this by calling `setKeyboardFocusEntity(UNKNOWN_ENTITY_ID)` first?

**Pull request merged**